tests/test_transform_hoist_variables.py

```diff
@@ -165,15 +166,15 @@ def test_hoist_disable(here, frontend, config):
     thus hoisting all (non-parameter) local variables for the non-disabled subroutines.
     """
 
+    disable = ("device1", "device2")
+    config['routine'].append({'name': 'kernel2', 'role': 'kernel', 'ignore': disable})
     proj = here/'sources/projHoist'
     scheduler = Scheduler(paths=[proj], config=config, seed_routines=['driver', 'another_driver'], frontend=frontend)
 
-    disable = ("device1", "device2")
-
     # Transformation: Analysis
-    scheduler.process(transformation=HoistVariablesAnalysis(disable=disable), reverse=True)
+    scheduler.process(transformation=HoistVariablesAnalysis(), reverse=True)
```

Comment:
[No action] Coming to think of it, I believe we wouldn't actually need the separate passes for analysis and transformation but could directly apply the hoisting via a reverse traversal.
On the other hand, I wonder whether it wouldn't be beneficial to create a more generic analysis pass which collects just information about temporaries per routine and recursively for the call tree, and then use this for both hoisting and pool allocator. But that's a separate PR.

Reply:
Agreed, a generic temporary analysis could be useful in more than one pass; good idea for the future, but beyond this PR.